core/src/avm1/scope.rs

```diff
@@ -30,10 +29,10 @@ pub enum ScopeClass {
 }
 
 /// Represents a scope chain for an AVM1 activation.
-#[derive(Debug, Collect)]
+#[derive(Copy, Clone, Debug, Collect)]
 #[collect(no_drop)]
 pub struct Scope<'gc> {
-    parent: Option<GcCell<'gc, Scope<'gc>>>,
+    parent: Option<Gc<'gc, Scope<'gc>>>,
     class: ScopeClass,
     values: Object<'gc>,
```

Comment:
Might not be related, but I think `values` should be reduced to just `PropertyMap<'gc, Value<'gc>>`. `Object<'gc>` includes many functionality that doesn't make sense in the context of `Scope` (e.g. prototype, watchers, interfaces etc.). Though this might belong to a separate PR.

Reply:
What about `with` blocks? Don't they trigger watchers and all that jazz?

Reply:
Since this isn't really related to this PR, let's defer the discussion to another time (which I don't have currently).